tools/rosbag/src/recorder.cpp

```diff
@@ -347,7 +347,11 @@ void Recorder::doQueue(const ros::MessageEvent<topic_tools::ShapeShifter const>&
                 ros::M_string::const_iterator it2 = out.connection_header->find("callerid");
                 if (it2 != out.connection_header->end())
                 {
-                    latched_msgs_.insert({{subscriber->getTopic(), it2->second}, out});
+                    auto const result = latched_msgs_.insert({{subscriber->getTopic(), it2->second}, out});
+                    if (not result.second)
```

Comment:
What do you think of adding a comment briefly explaining why this is required?

Reply:
Sure, that makes sense.

Reply:
I added a not regarding the behavior of the map::insert function. I did not want to write too much to be consistent with the rest of the code. Please let me know if this makes sense.